driver.go

```diff
@@ -13,7 +14,9 @@ import (
 type Driver interface {
 	// Connect connects to the database. It does not need to call `db.Ping()`
 	Connect(backend.DataSourceInstanceSettings) (*sql.DB, error)
+	Timeout(backend.DataSourceInstanceSettings) time.Duration
```

Comment:
Do we need this to be explicit like `QueryTimeout`? We may have to add `ConnectionTimeout` in the future.

Reply:
Actually that gives me another idea. Good call though